pkg/test/framework/components/prometheus/prometheus.go

```diff
@@ -26,30 +26,30 @@ type Instance interface {
 	resource.Resource
 
 	// API Returns the core Prometheus APIs.
-	API() v1.API
+	API() []v1.API
```

Comment:
Can we make the interface return `APIForCluster(resource.Cluster)`

And probably move all of the methods below to another interface and have a getter `PromForCluster`